include/zephyr/pm/device.h

```diff
@@ -231,21 +302,26 @@ struct pm_device {
  * @param dev_id Device id.
  * @param pm_action_cb PM control callback.
  */
-#define Z_PM_DEVICE_DEFINE(node_id, dev_id, pm_action_cb)		\
-	Z_PM_DEVICE_DEFINE_SLOT(dev_id);				\
-	static struct pm_device Z_PM_DEVICE_NAME(dev_id) =		\
-	Z_PM_DEVICE_INIT(Z_PM_DEVICE_NAME(dev_id), node_id,		\
-			 pm_action_cb)
+#define Z_PM_DEVICE_DEFINE(node_id, dev_id, pm_action_cb)	\
+	Z_PM_DEVICE_DEFINE_SLOT(dev_id);			\
+	static struct pm_device_rt Z_PM_DEVICE_NAME(dev_id) =	\
+	Z_PM_DEVICE_INIT(Z_PM_DEVICE_NAME(dev_id), node_id, pm_action_cb)
+
+#define Z_PM_DEVICE_SYNC_DEFINE(node_id, dev_id, pm_action_cb)				\
```

Comment:
Can't we avoid additional macros ... 
Maybe  something around ...
```
#define PM_DEVICE_DEFINE(dev_id, pm_action_cb, ...) \
	COND_CODE_1(Z_IS_EMPTY(__VA_ARGS__), \
	(Z_PM_DEVICE_DEFINE(DT_INVALID_NODE, dev_id, pm_action_cb)), \
	(Z_PM_DEVICE_SYNC_DEFINE(DT_INVALID_NODE, dev_id, pm_action_cb)))
```

Reply:
Sounds good to me if we extend API macro `PM_DEVICE_DEFINE(node, dev, cb, ...)`  I am not sure what would be the additional argument. Should it be something like `PM_DEVICE_DEFINE(..., PM_DEVICE_TYPE_<name>)` where flag can indicate generic device or the new one and for backward compatibility no argument would mean the generic runtime pm?

Reply:
Extended that was. Optional flag can be provided to indicate ISR safe type.
`PM_DEVICE_DEVICE(dev, cb)` standard device
`PM_DEVICE_DEVICE(dev, cb, 0)` standard device
`PM_DEVICE_DEVICE(dev, cb, PM_DEVICE_ISR_SAFE)` ISR safe device

`PM_DEVICE_ISR_SAFE` is just defined as 1.